stubs/pyOpenSSL/OpenSSL/SSL.pyi

```diff
@@ -115,25 +117,63 @@ def SSLeay_version(type: int) -> str: ...
 class Session: ...
 
 class Connection:
-    def __getattr__(self, name: str) -> Any: ...  # incomplete
+    def __getattr__(self, name: str) -> Any: ...  # takes attributes from `self._socket`
     def __init__(self, context: Context, socket: socket.socket | None = ...) -> None: ...
+    def get_context(self) -> Context: ...
+    def set_context(self, context: Context) -> None: ...
+    def get_servername(self) -> bytes | None: ...
+    def set_tlsext_host_name(self, name: bytes) -> None: ...
+    def pending(self) -> int: ...
+    def send(self, buf: ReadableBuffer, flags: int = ...) -> int: ...
+    write = send
+    def sendall(self, buf: ReadableBuffer, flags: int = ...) -> int: ...
+    def recv(self, bufsiz: int, flags: int | None = ...) -> bytes: ...
+    read = recv
+    def recv_into(self, buffer: bytearray, nbytes: int | None = ..., flags: int | None = ...) -> int: ...
     def connect(self, addr: str | bytes | Sequence[str | int]) -> None: ...
+    def connect_ex(self, addr: _Address | bytes) -> int: ...
+    def accept(self) -> tuple[Connection, _RetAddress]: ...
+    def shutdown(self) -> bool: ...
     def do_handshake(self) -> None: ...
-    def get_peer_certificate(self) -> X509: ...
-    def set_tlsext_host_name(self, name: bytes) -> None: ...
+    def get_certificate(self) -> X509 | None: ...
+    def get_peer_certificate(self) -> X509 | None: ...
+    def get_peer_cert_chain(self) -> list[X509] | None: ...
+    def get_verified_chain(self) -> list[X509] | None: ...
     def bio_read(self, bufsiz: int) -> bytes: ...
     def bio_write(self, buf: bytes) -> int: ...
-    def recv(self, bufsiz: int, flags: int | None = ...) -> bytes: ...
-    def sendall(self, buf: bytes, flags: int = ...) -> int: ...
+    def bio_shutdown(self) -> None: ...
+    def renegotiate(self) -> bool: ...
+    def renegotiate_pending(self) -> bool: ...
+    def total_renegotiations(self) -> int: ...
+    def makefile(self, *args: Any, **kwargs: Any) -> NoReturn: ...
```

Comment:
Maybe it would be better to leave this one out, since it just unconditionally raises an exception?

```suggestion
```

Reply:
Makes sense!